anycloud/cli/src/deploy/mod.rs

```diff
@@ -1316,13 +1316,15 @@ where
     // update the spinner and lines above the spinner
     new_lines
       .into_iter()
-      .filter(|line| !line.is_empty())
-      .for_each(|line| {
-        if let Some(last_line) = lines.get(lines.len() - 1) {
-          sp.println(last_line);
+      .filter(|new_line| !new_line.is_empty())
+      .for_each(|new_line| {
+        if lines.len() > 0 {
+          if let Some(last_line) = lines.get(lines.len() - 1) {
+            sp.println(last_line);
+          }
         }
-        sp.set_message(line);
-        lines.push(line.to_string());
+        sp.set_message(new_line);
+        lines.push(new_line.to_string());
       });
```

Comment:
The issue with this is if `new_line` has multiple new lines compared to the old `lines` it will be printed multiple times.

It would be better to break this out of the `for_each` or somehow make sure it's only run on the first iteration of that loop?